build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,10 +173,62 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] =
+					details.releaseGroup === "client"
+						? details.ranges.legacyCompat
+						: details.ranges.caret;
+			}
+			break;
+		}
+
 		default: {
 			throw new Error(`Unexpected ReportKind: ${kind}`);
 		}
 	}
 
 	return toReturn;
 }
+
+/**
+ * Generates a new semantic version representing the next version in a legacy compatibility range based on a specified multiple of minor versions.
+ *
+ * This function returns the minor version of the given version to the nearest  multiple of `compatVersionInterval` and bumps it by the `compatVersionInterval` to generate
+ * a new semantic version.
+ *
+ * @param version - A semver-compatible string or `semver.SemVer` object representing the current version.
+ * @param compatVersionInterval - The multiple of minor versions to use for calculating the next version in the range.
+ *
+ * @returns A new `semver.SemVer` object representing the next version in the legacy compatibility range.
+ */
+export function getInternalVersionRange(
+	version: semver.SemVer | string,
+	compatVersionInterval: number,
+): string {
```

Comment:
docs say this returns a semver, not a string. I think semver is probably better. (I have learned since writing this code that string arguments in function calls are quite challenging maintenance wise, so when you have the option to use something else, it's often worth doing.)

Reply:
using `semver.parse(v)` with a version range as input will result in null, as the function parse only valid semantic version strings, not ranges.

Reply:
Ahh, got it. I think the docs just need to be updated then.